src/clj/puppetlabs/services/master/master_service.clj

```diff
@@ -53,36 +80,31 @@
-     (let [route-config (core/get-master-route-config ::master-service config)
-           path (core/get-master-mount ::master-service route-config)
-           ring-app (core/construct-root-routes puppet-version
-                                                use-legacy-auth-conf
-                                                jruby-service
-                                                get-code-content
-                                                handle-request
-                                                (get-auth-handler)
-                                                environment-class-cache-enabled)
-           routes (comidi/context path ring-app)
-           comidi-handler (comidi/routes->handler routes)
-           ring-handler (comidi/wrap-with-route-metadata comidi-handler routes)]
-       ;; if the webrouting config uses the old-style config where
-       ;; there is a single key with a route-id, we need to deal with that
-       ;; for backward compat.  We have a hard-coded assumption that this route-id
-       ;; must be `master-routes`.  In Puppet Server 2.0, we also supported a
-       ;; key called `invalid-in-puppet-4` in the same route config, even though
-       ;; that key is no longer used for Puppet Server 2.1 and later.  We
-       ;; should be able to remove this hack as soon as we are able to get rid
-       ;; of the legacy routes.
-       (if (and (map? route-config)
-                (contains? route-config :master-routes))
-         (add-ring-handler this
-                           ring-handler
-                           {:route-id :master-routes
-                            :normalize-request-uri true})
-         (add-ring-handler this
-                           ring-handler
-                           {:normalize-request-uri true}))))
-   context)
+
+     ;; if the webrouting config uses the old-style config where
+     ;; there is a single key with a route-id, we need to deal with that
+     ;; for backward compat.  We have a hard-coded assumption that this route-id
+     ;; must be `master-routes`.  In Puppet Server 2.0, we also supported a
+     ;; key called `invalid-in-puppet-4` in the same route config, even though
+     ;; that key is no longer used for Puppet Server 2.1 and later.  We
+     ;; should be able to remove this hack as soon as we are able to get rid
+     ;; of the legacy routes.
+     (if (and (map? route-config)
+              (contains? route-config :master-routes))
+       (add-ring-handler this
+                         ring-handler
+                         {:route-id :master-routes
+                          :normalize-request-uri true})
+       (add-ring-handler this
+                         ring-handler
+                         {:normalize-request-uri true}))
+
+     (register-status
+      "master"
+      (status-core/get-artifact-version "puppetlabs" "puppetserver")
+      1
+      (partial core/v1-status metrics))
```

Comment:
Hm... what if rather than duplicating this in the master service in pe-p-s-e, we just registered another status here with `"pe-master"`. Could have registering it dependent on whether the pe-p-s-e artifact is being used. Something like:

```clojure
    (let [version (versioneer/get-version "puppetlabs" "pe-puppet-server-extensions")]
      (when-not (empty? version)
        (register-status
         "pe-master"
         version
         1
         (fn [level]
           (log/warn "The pe-master status is deprecated")
           (core/v1-status metrics-registry level)))))
```

(`status-core/get-artifact-version` throws an error if it doesn't get back a version, which is why I put `versioneer/get-version` there.)